compiler/src/dotty/tools/dotc/transform/PatternMatcher.scala

```diff
@@ -326,7 +326,12 @@ object PatternMatcher {
           sym.isAllOf(SyntheticCase) && sym.owner.is(Scala2x)
 
         if (isSyntheticScala2Unapply(unapp.symbol) && caseAccessors.length == args.length)
-          matchArgsPlan(caseAccessors.map(ref(scrutinee).select(_)), args, onSuccess)
+          def tupleSel(sym: Symbol) = ref(scrutinee).select(sym)
+          def tupleApp(i: Int) = // manually inlining the call to NonEmptyTuple#apply, because it's an inline method
+            ref(defn.RuntimeTuplesModule).select(defn.RuntimeTuples_apply).appliedTo(ref(scrutinee), Literal(Constant(i)))
```

Comment:
The result of the `runtime.Tuples.apply` must be casted the type of the element

Reply:
This is the root of the failures in `FromExpr`

Reply:
I belive it should be
```suggestion
            ref(defn.RuntimeTuplesModule).select(defn.RuntimeTuples_apply).appliedTo(ref(scrutinee), Literal(Constant(i))).asInstance(args(i).tpe)
```